ui/apps/platform/cypress/integration/vulnerabilities/exceptionManagement/denyRequestFlow.test.ts

```diff
@@ -0,0 +1,84 @@
+import withAuth from '../../../helpers/basicAuth';
+import { hasFeatureFlag } from '../../../helpers/features';
+import { getInputByLabel } from '../../../helpers/formHelpers';
+import { cancelAllCveExceptions } from '../workloadCves/WorkloadCves.helpers';
+import { deferAndVisitRequestDetails } from './ExceptionManagement.helpers';
+
+const comment = 'Defer me';
+const expiry = 'When all CVEs are fixable';
+const scope = 'All images';
+
+describe('Exception Management Request Details Page', () => {
+    withAuth();
+
+    before(function () {
+        if (
+            !hasFeatureFlag('ROX_VULN_MGMT_WORKLOAD_CVES') ||
+            !hasFeatureFlag('ROX_VULN_MGMT_UNIFIED_CVE_DEFERRAL') ||
+            !hasFeatureFlag('ROX_WORKLOAD_CVES_FIXABILITY_FILTERS')
+        ) {
+            this.skip();
+        }
+    });
+
+    beforeEach(() => {
+        if (
+            hasFeatureFlag('ROX_VULN_MGMT_WORKLOAD_CVES') &&
+            hasFeatureFlag('ROX_VULN_MGMT_UNIFIED_CVE_DEFERRAL') &&
+            hasFeatureFlag('ROX_WORKLOAD_CVES_FIXABILITY_FILTERS')
+        ) {
+            cancelAllCveExceptions();
+            deferAndVisitRequestDetails({
+                comment,
+                expiry,
+                scope,
+            });
+        }
+    });
+
+    after(() => {
+        if (
+            hasFeatureFlag('ROX_VULN_MGMT_WORKLOAD_CVES') &&
+            hasFeatureFlag('ROX_VULN_MGMT_UNIFIED_CVE_DEFERRAL') &&
+            hasFeatureFlag('ROX_WORKLOAD_CVES_FIXABILITY_FILTERS')
+        ) {
+            cancelAllCveExceptions();
+        }
+    });
+
+    it('should be able to deny a request if approval permissions are granted', () => {
+        cy.get('button:contains("Deny request")').click();
+        cy.get('div[role="dialog"]').should('exist');
+        getInputByLabel('Denial rationale').type('Denied');
+        cy.get('div[role="dialog"] button:contains("Deny")').click();
+        cy.get('div[role="dialog"]').should('not.exist');
+        cy.get('div[aria-label="Success Alert"]').should(
+            'contain',
+            'The vulnerability request was successfully denied.'
+        );
+    });
+
+    it('should be able to see how many CVEs will be affected by a denial', () => {
+        cy.get('table tbody tr:not(".pf-c-table__expandable-row")').then((rows) => {
+            const numCVEs = rows.length;
+            cy.get('button:contains("Deny request")').click();
+            cy.get('div[role="dialog"]').should('exist');
+            cy.get(`div:contains("CVE count: ${numCVEs}")`).should('exist');
+        });
```

Comment:
My initial thought with this was that it was susceptible to a table with more than one page of CVEs, but given the way the tests are structured that should never be a problem. One thing that we _do_ need as a separate task though is to add pagination controls to this CVE table. Currently if there is a request with > 20 CVEs, it will not be possible to view CVEs beyond the first page.

Reply:
That's a good point. The pagination might have been overlooked. I can add a ticket to add that in a future PR